File: backend/assessmentservice/src/main/java/com/openclassrooms/assessmentservice/model/RiskLevel.java
package com.openclassrooms.assessmentservice.model;

import lombok.Getter;

/**
 * Niveaux de risque de diabète selon les règles métier.
 *
 * @author Kardigué MAGASSA
 * @version 1.0
 * @since 2026-02-09
 */
@Getter
public enum RiskLevel {
    NONE("Aucun risque"),
    BORDERLINE("Risque limité"),
    IN_DANGER("Danger"),
    EARLY_ONSET("Apparition précoce"),
    UNKNOWN("Inconnu");

    private final String description;

    RiskLevel(String description) {this.description = description;}

}

File: frontend/mediLabo-solutions-ui/src/app/enum/riskLevel.ts
export enum RiskLevel {
  NONE = 'NONE',
  BORDERLINE = 'BORDERLINE',
  IN_DANGER = 'IN_DANGER',
  EARLY_ONSET = 'EARLY_ONSET',
  UNKNOWN = 'UNKNOWN'
};

export const RiskLevelUI = {
  [RiskLevel.NONE]: {label: 'Aucun risque', color: 'green'},
  [RiskLevel.BORDERLINE]: {label: 'Risque limité',color: 'orange'},
  [RiskLevel.IN_DANGER]: {label: 'Danger',color: 'red'},
  [RiskLevel.EARLY_ONSET]: {label: 'Apparition précoce',color: 'purple'},
  [RiskLevel.UNKNOWN]: {label: 'Inconnu',color: 'gray'}
};